python/pip_install/extract_wheels/lib/wheel.py

```diff
@@ -37,57 +37,46 @@ def path(self) -> str:
+            metadata_contents = wheel_source.read_dist_info("METADATA")
+            metadata = installer.utils.parse_metadata_file(metadata_contents)
+        return metadata
 
-    def entry_points(self) -> Dict[str, str]:
+    @property
+    def version(self) -> str:
+        # TODO Also available as installer.sources.WheelSource.version
+        return str(self.metadata["Version"])
+
+    def entry_points(self) -> Dict[str, tuple[str, str]]:
         """Returns the entrypoints defined in the current wheel
 
         See https://packaging.python.org/specifications/entry-points/ for more info
 
         Returns:
-            Dict[str, str]: A mappying of the entry point's name to it's method
+            Dict[str, Tuple[str, str]]: A mapping of the entry point's name to it's module and attribute
         """
-        with zipfile.ZipFile(self.path, "r") as whl:
-            # Calculate the location of the entry_points.txt file
-            metadata = self.metadata
-            name = "{}-{}".format(metadata.name.replace("-", "_"), metadata.version)
-
-            # Note that the zipfile module always uses the forward slash as
-            # directory separator, even on Windows, so don't use os.path.join
-            # here.  Reference for Python 3.10:
-            # https://github.com/python/cpython/blob/3.10/Lib/zipfile.py#L355.
-            # TODO: use zipfile.Path once 3.8 is our minimum supported version
-            entry_points_path = "{}.dist-info/entry_points.txt".format(name)
-
-            # If this file does not exist in the wheel, there are no entry points
-            if entry_points_path not in whl.namelist():
+        with installer.sources.WheelFile.open(self.path) as wheel_source:
+            if "entry_points.txt" not in wheel_source.dist_info_filenames:
                 return dict()
 
-            # Parse the avaialble entry points
-            config = configparser.ConfigParser()
-            try:
-                config.read_string(whl.read(entry_points_path).decode("utf-8"))
-                if "console_scripts" in config.sections():
-                    return dict(config["console_scripts"])
-
-            # TODO: It's unclear what to do in a situation with duplicate sections or options.
-            # For now, we treat the config file as though it contains no scripts. For more
-            # details on the config parser, see:
-            # https://docs.python.org/3.7/library/configparser.html#configparser.ConfigParser
-            # https://docs.python.org/3.7/library/configparser.html#configparser.Error
-            except configparser.Error:
-                pass
+            entry_points_mapping = dict()
+            entry_points_contents = wheel_source.read_dist_info("entry_points.txt")
+            entry_points = installer.utils.parse_entrypoints(entry_points_contents)
+            for script, module, attribute, kind in entry_points:
+                if kind == "console":
```

Comment:
The parse_entrypoints function says this is a `"ScriptSection"`, but `kind` is maybe easier to grok than that.